build-common.xml

```diff
@@ -37,15 +37,15 @@
             <fileset dir="${ivy.jar.dir}" includes="*.jar"/>
         </path>
         <taskdef resource="org/apache/ivy/ant/antlib.xml" uri="antlib:org.apache.ivy.ant" classpathref="ivy.lib.path"/>
-        <ivy:settings id="dev.settings" file="../build-ivysettings.xml"/>
     </target>
     <!-- common resolve target that should work to resolve dependencies based on ivy.xml for most projects -->
     <target name="resolve" depends="init-ivy" description="resolve dependencies">
+      <ivy:settings id="dev.settings" file="../build-ivysettings.xml"/>
       <ivy:resolve settingsRef="dev.settings" />
-      <ivy:cachepath pathid="class.path" />
     </target>
 
-    <target name="taskdefs"  description="Declare 3rd party ANT tasks" depends="resolve">
+    <target name="taskdefs"  description="Declare 3rd party ANT tasks">
```

Comment:
`taskdefs` doesn't intuitively shout "tasks provided by antcontrib" to me.  How about renaming `def-antcontrib-tasks`?  Happy to be over-ridden on this if others thing that is too wordy.